ibis/expr/api.py

```diff
@@ -14,53 +14,31 @@
 
 from __future__ import print_function
 
+import six
+import toolz
 import warnings
 import operator
 import datetime
 import functools
 import collections
 
-import six
-import toolz
+import ibis.util as util
+import ibis.common as com
+import ibis.expr.types as ir
+import ibis.expr.rules as rlz
+import ibis.expr.schema as sch
+import ibis.expr.analysis as _L
+import ibis.expr.datatypes as dt
+import ibis.expr.analytics as _analytics
+import ibis.expr.operations as ops
 
-from ibis.expr.schema import Schema
-from ibis.expr import datatypes as dt
-from ibis.expr import schema as sch
-from ibis.expr.types import (Expr,  # noqa
-                             ValueExpr, ScalarExpr, ColumnExpr,
-                             TableExpr,
-                             NumericValue, NumericColumn,
-                             IntegerValue,
-                             Int8Value, Int8Scalar, Int8Column,
-                             Int16Value, Int16Scalar, Int16Column,
-                             Int32Value, Int32Scalar, Int32Column,
-                             Int64Value, Int64Scalar, Int64Column,
-                             NullScalar,
-                             BooleanValue, BooleanScalar, BooleanColumn,
-                             FloatingValue,
-                             FloatValue, FloatScalar, FloatColumn,
-                             DoubleValue, DoubleScalar, DoubleColumn,
-                             StringValue, StringScalar, StringColumn,
-                             DecimalValue, DecimalScalar, DecimalColumn,
-                             TimestampValue, TimestampScalar, TimestampColumn,
-                             IntervalValue, IntervalScalar, IntervalColumn,
-                             DateValue, TimeValue,
-                             ArrayValue, ArrayScalar, ArrayColumn,
-                             MapValue, MapScalar, MapColumn,
-                             StructValue, StructScalar, StructColumn,
-                             CategoryValue, unnamed, as_value_expr, literal,
-                             param, null, sequence)
-
-import ibis.common as _com
 from ibis.compat import PY2, to_time, to_date
+from ibis.expr.types import Expr, null, param, literal, sequence, as_value_expr
```

Comment:
We might factor out `null`, `param`, `literal`, `sequence`, `as_value_expr` to here, but we can do that incrementally :)